'Services' > 'Settings' section

carma12 opened the pull request:
![image](https://user-images.githubusercontent.com/8112750/221526215-39ee0e00-2135-4cb4-ab9d-8e53bd9b8422.png)

### Description
The 'Services' > 'Settings' section contains the main configuration for a specific service (accessed by clicking on the main table). This can be reached under `<BASE-URL>/services/settings`.

### Routing
The routing has been defined in the `AppRoutes` component to access this page in the WebUI. As this belongs to the Service section, it should depend on it. The routing points directly to the `ServicesSettings` component.

```ts
<Route path="services">
   <Route path="" element={<Services />} />
   <Route path="settings" element={<ServicesTabs />} />
</Route>
```

### 'Settings' section
The Services' Settings page (`ServicesSettings` component) contains the following subsections:

- 'Service settings' (`ServiceSettings` component)
   - Related to the settings of a specific service.
- 'Provisioning' (`Provisioning` component)
   - Related to Kerberos authentication settings.
- 'Service certificate' (`ServiceCertificate` component)
   - Related to the certificates associated with that specific host.
- 'Allow to retrieve keytab' (`AllowedRetrieveKeytab` component)
   - Related to the permissions for retrieving keytabs.
- 'Allow to create keyTab' (`AllowedCreateKeytab` component)
   - Related to the permissions for creating keytabs.

Also, the `ServicesSettings` component has been set under a layout called `ServicesTabs`. This structure makes the code leaner and more scalable.

**NOTE:** The `AllowedRetrieveKeytab` and `AllowedCreateKeytab` components have been created specifically to be used by the `Service` data type. But in turn, those are reusing other components that have been used in `HostSettings` (the tables to represent the different entities: 'User', 'Host', 'User group', etc). This needs to be adapted in the future to be used with `Host` and `Service` data types indistinctively.

Signed-off-by: Carla Martinez <carlmart@redhat.com>

abbra commented:
@carma12 from the screenshot I see that `idp` authentication indicator is missing:

```
        StrEnum(
            'krbprincipalauthind*',
            cli_name='auth_ind',
            label=_('Authentication Indicators'),
            doc=_("Defines an allow list for Authentication Indicators."
                  " Use 'otp' to allow OTP-based 2FA authentications."
                  " Use 'radius' to allow RADIUS-based 2FA authentications."
                  " Use 'pkinit' to allow PKINIT-based 2FA authentications."
                  " Use 'hardened' to allow brute-force hardened password"
                  " authentication by SPAKE or FAST."
                  " With no indicator specified,"
                  " all authentication mechanisms are allowed."),
            values=(u'radius', u'otp', u'pkinit', u'hardened', u'idp'),
        ),

```

We will need to update the doc string upstream (see `ipaserver/plugins/service.py:class serviceLDAPObject)`) to add `idp` string there too.

carma12 commented:
@abbra Thanks for pointing that. I have already corrected the code and updated the image in the description with the fix. Also, I have created a [new issue](https://github.com/freeipa/freeipa-webui/issues/58) to track the updates of the doc string for the new WebUI.

pvoborni commented on src/components/ServicesSections/ServiceCertificate.tsx:
```diff
@@ -0,0 +1,41 @@
+    <Flex direction={{ default: "column", lg: "row" }}>
+      <FlexItem flex={{ default: "flex_1" }}>
+        <Form className="pf-u-mb-lg">
+          <FormGroup label="Certificates" fieldId="certificates">
```
The logic for certificates is the same for user, host and service objects.  We could have a single component which handles it for all the pages.

pvoborni reviewed (changes requested):
Finished first iteration.  Found some typos. Mostly looks good. But I'd like to discuss approach to code repetition/DRY principal on our mtg.

carma12 commented on src/components/ServicesSections/AllowedCreateKeytab.tsx:
```diff
@@ -0,0 +1,31 @@
+  return (
+    <Flex direction={{ default: "column", lg: "row" }}>
+      <FlexItem flex={{ default: "flex_1" }}>
+        <CreateKeytabUsersTable host={props.service.id} />
```
Good idea. Also, I have to do the same thing in many components of the project... Those are easy to fix but require to be done in different PRs to keep consistency.

I have created a [new issue](https://github.com/freeipa/freeipa-webui/issues/65) for this one to keep track of it.

carma12 commented:
Thanks for the review. I have already fixed the typo issues and added the topics related to the optimization/creation of some components to our agenda to discuss in our meetings.

pvoborni reviewed (approved):
The PR doesn't break stuff thus approving. But in long run, we should focus on the comments about code reuse I made.
